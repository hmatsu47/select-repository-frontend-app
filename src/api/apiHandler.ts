import { fetchWithTimeout } from "./fetchWithTimeout";

// ベース URI
export const baseUri = "http://10.11.17.43/api";

// APIからデータ取得（GET）
export const getApiData = async (url: string) => {
  return await (await fetchWithTimeout(url)).json();
};
// APIからデータ取得（POST）
export const postApiData = async (url: string, object: Object) => {
  return await (await fetchWithTimeout(url, object)).json();
};
