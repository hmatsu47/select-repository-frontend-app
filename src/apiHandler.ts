import { fetchWithTimeout } from "./fetchWithTimeout";

// APIからデータ取得（GET）
export const getApiData = async (url: string) => {
  return await (await fetchWithTimeout(url)).json();
};
// APIからデータ取得（POST）
export const postApiData = async (url: string, object: Object) => {
  return await (await fetchWithTimeout(url, object)).json();
};
