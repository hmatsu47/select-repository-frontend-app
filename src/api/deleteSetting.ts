import { ErrorResponse } from "../type";
import { baseUri, deleteApiData } from "./apiHandler";
import {
  imageUri,
  isReleased,
  isReleaseSelected,
  releaseAt,
  service,
  setMessage,
  setMessageSeverity,
  setImageUri,
  setIsReleaseSelected,
  setReleaseAt,
} from "../signal";

export const deleteSetting = async () => {
  if (
    !isReleased() === undefined ||
    !isReleaseSelected() ||
    !imageUri() ||
    !releaseAt()
  ) {
    return;
  }
  const load = async (): Promise<void> => {
    const data: null | ErrorResponse = await deleteApiData(
      `${baseUri}/setting/${service()}`
    );
    if (
      typeof data === "object" &&
      data !== null &&
      typeof (data as ErrorResponse).message === "string"
    ) {
      // 戻り値がエラーメッセージの場合
      setMessage((data as ErrorResponse).message);
      setMessageSeverity("error");
      return;
    }
    setImageUri(undefined);
    setReleaseAt(undefined);
    setIsReleaseSelected(false);
    setMessage("リリースを取り消しました。");
    setMessageSeverity("success");
  };
  void load();
};
