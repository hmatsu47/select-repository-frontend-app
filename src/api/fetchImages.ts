import { ErrorResponse, ImageItem } from "../type";
import { baseUri, getApiData } from "./apiHandler";
import { repository, service, setErrorMessage, setImages } from "../signal";

export const fetchImages = async () => {
  if (!service() || !repository()) {
    return;
  }
  const load = async (): Promise<void> => {
    const data: ImageItem[] | ErrorResponse = await getApiData(
      `${baseUri}/images/${service()}/${repository()}`
    );
    if (
      typeof data === "object" &&
      data !== null &&
      typeof (data as ErrorResponse).message === "string"
    ) {
      // 戻り値がエラーメッセージの場合
      setImages(undefined);
      setErrorMessage((data as ErrorResponse).message);
      return;
    }
    setImages(data as ImageItem[]);
    setErrorMessage(undefined);
  };
  void load();
};
