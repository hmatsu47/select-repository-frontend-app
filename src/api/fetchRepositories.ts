import { ErrorResponse, RepositoryItem } from "../type";
import { baseUri, getApiData } from "./apiHandler";
import {
  service,
  setImages,
  setMessage,
  setMessageSeverity,
  setRepositories,
} from "../signal";

export const fetchRepositories = async () => {
  if (!service()) {
    return;
  }
  const load = async (): Promise<void> => {
    const data: RepositoryItem[] | ErrorResponse = await getApiData(
      `${baseUri}/repositories/${service()}`
    );
    if (
      typeof data === "object" &&
      data !== null &&
      typeof (data as ErrorResponse).message === "string"
    ) {
      // 戻り値がエラーメッセージの場合
      setRepositories(undefined);
      setImages(undefined);
      setMessage((data as ErrorResponse).message);
      setMessageSeverity("error");
      return;
    }
    setRepositories(data as RepositoryItem[]);
    setImages(undefined);
    setMessage(undefined);
  };
  void load();
};
