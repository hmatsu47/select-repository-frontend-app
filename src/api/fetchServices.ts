import { ErrorResponse, ServiceItem } from "../type";
import { baseUri, getApiData } from "./apiHandler";
import {
  setErrorMessage,
  setImages,
  setRepositories,
  setServices,
} from "../signal";

export const fetchServices = async () => {
  const load = async (): Promise<void> => {
    const data: ServiceItem[] | ErrorResponse = await getApiData(
      `${baseUri}/services`
    );
    if (
      typeof data === "object" &&
      data !== null &&
      typeof (data as ErrorResponse).message === "string"
    ) {
      // 戻り値がエラーメッセージの場合
      setServices(undefined);
      setRepositories(undefined);
      setImages(undefined);
      setErrorMessage((data as ErrorResponse).message);
      return;
    }
    setServices(data as ServiceItem[]);
    setRepositories(undefined);
    setImages(undefined);
    setErrorMessage(undefined);
  };
  void load();
};
