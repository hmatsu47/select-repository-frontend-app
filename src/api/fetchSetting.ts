import { ErrorResponse, Setting } from "../type";
import { baseUri, getApiData } from "./apiHandler";
import {
  service,
  setErrorMessage,
  setImageUri,
  setIsReleased,
  setReleaseAt,
} from "../signal";

export const fetchSetting = async () => {
  if (!service()) {
    return;
  }
  const load = async (): Promise<void> => {
    const data: Setting | ErrorResponse = await getApiData(
      `${baseUri}/setting/${service()}`
    );
    if (
      typeof data === "object" &&
      data !== null &&
      typeof (data as ErrorResponse).message === "string"
    ) {
      // 戻り値がエラーメッセージの場合
      setIsReleased(undefined);
      setImageUri(undefined);
      setReleaseAt(undefined);
      setErrorMessage((data as ErrorResponse).message);
      return;
    }
    const tmpIsReleased = (data as Setting).is_released;
    setIsReleased(tmpIsReleased);
    setImageUri(tmpIsReleased ? (data as Setting).image_uri : undefined);
    setReleaseAt(tmpIsReleased ? (data as Setting).release_at : undefined);
    setErrorMessage(undefined);
  };
  void load();
};
