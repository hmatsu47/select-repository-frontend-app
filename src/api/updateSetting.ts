import { ErrorResponse, Setting } from "../type";
import { baseUri, postApiData } from "./apiHandler";
import {
  imageUri,
  isReleased,
  releaseAt,
  service,
  setMessage,
  setMessageSeverity,
} from "../signal";

export const updateSetting = async () => {
  if (!isReleased() === undefined || !imageUri() || !releaseAt()) {
    return;
  }
  const load = async (): Promise<void> => {
    const data: Setting | ErrorResponse = await postApiData(
      `${baseUri}/setting/${service()}`,
      {
        image_uri: imageUri(),
        is_released: isReleased(), // 値は無視される
        release_at: releaseAt(),
      } as Setting
    );
    if (
      typeof data === "object" &&
      data !== null &&
      typeof (data as ErrorResponse).message === "string"
    ) {
      // 戻り値がエラーメッセージの場合
      setMessage((data as ErrorResponse).message);
      setMessageSeverity("error");
      return;
    }
    setMessage("リリースイメージ URI とリリース日時をセットしました。");
    setMessageSeverity("success");
  };
  void load();
};
