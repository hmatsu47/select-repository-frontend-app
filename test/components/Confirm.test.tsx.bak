// @vitest-environment jsdom

import { describe, expect, test, vi } from "vitest";
import "vi-fetch/setup";
import { mockFetch, mockPost } from "vi-fetch";
import { fireEvent, render } from "solid-testing-library";
import { formatSnapshot } from "../common/formatSnapshot";
import { Confirm } from "../../src/components/Confirm";
import { baseUri } from "../../src/api/apiHandler";
import { updateSetting } from "../../src/api/updateSetting";
import {
  imageUri,
  isOpenedConfirm,
  isReleased,
  releaseAt,
  service,
  setImageUri,
  setIsOpenedConfirm,
  setIsReleased,
  setReleaseAt,
  setService,
} from "../../src/signal";
import { Setting } from "../../src/type";

describe("<Confirm />", () => {
  const confirmList = [
    {
      title: "はい・過去リリースあり",
      buttonYes: true,
      service: "hoge",
      isReleased: true,
      imageUri: "000000000000.dkr.ecr.ap-northeast-1.amazonaws.com/hoge:latest",
      releaseAt: new Date(2022, 0, 1, 0, 0),
      testFetch: true,
    },
    {
      title: "はい・過去リリースなし",
      buttonYes: true,
      service: "hoge",
      isReleased: false,
      imageUri: "000000000000.dkr.ecr.ap-northeast-1.amazonaws.com/hoge:latest",
      releaseAt: new Date(2022, 0, 1, 0, 0),
      testFetch: true,
    },
    {
      title: "いいえ・過去リリースあり",
      buttonYes: false,
      service: "hoge",
      isReleased: true,
      imageUri: "000000000000.dkr.ecr.ap-northeast-1.amazonaws.com/hoge:latest",
      releaseAt: new Date(2022, 0, 1, 0, 0),
      testFetch: false,
    },
  ];
  beforeEach(() => {
    mockFetch.clearAll();
  });
  afterEach(() => {
    vi.clearAllMocks();
  });
  confirmList.forEach((testCase) => {
    test(testCase.title, async () => {
      setService(testCase.service);
      setIsReleased(testCase.isReleased);
      setImageUri(testCase.imageUri);
      setReleaseAt(testCase.releaseAt);
      setIsOpenedConfirm(true);
      const mock = mockPost(`${baseUri}/setting/${service()}`)
        .withHeaders([["Content-Type", "application/json;charset=utf8"]])
        .willResolve({
          image_uri: testCase.imageUri,
          is_released: testCase.isReleased,
          release_at: testCase.releaseAt,
        } as Setting);
      await updateSetting();
      const { findByTitle, unmount } = render(() => <Confirm />);
      // はいボタン
      const buttonYes = (await findByTitle("はい")) as HTMLInputElement;
      expect(buttonYes).toHaveTextContent("はい");
      // いいえボタン
      const buttonNo = (await findByTitle("いいえ")) as HTMLInputElement;
      expect(buttonNo).toHaveTextContent("いいえ");
      const modal = (await findByTitle("modal")) as HTMLElement;
      // css の名前が動的に変わるので固定値に置換
      const htmlModal = formatSnapshot(modal.innerHTML);
      expect(htmlModal).toMatchSnapshot();
      // ボタンクリック
      fireEvent.click(testCase.buttonYes ? buttonYes : buttonNo);
      // API 呼び出しチェック
      if (testCase.testFetch) {
        // とりあえず呼び出しが行われたことだけを確認
        expect(mock).toHaveFetched();
        expect(mock).toHaveFetchedWithBody(
          JSON.stringify({
            image_uri: imageUri(),
            is_released: isReleased(),
            release_at: releaseAt(),
          } as Setting)
        );
      }
      // モーダルクローズ
      expect(isOpenedConfirm()).toBe(false);
      unmount();
    });
  });
});
