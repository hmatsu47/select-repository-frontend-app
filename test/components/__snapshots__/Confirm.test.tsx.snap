// Vitest Snapshot v1

exports[`<Confirm /> > いいえ・過去リリースあり 1`] = `"<h6 class=\\"MuiTypography-root MuiTypography-subtitle1 MuiTypography-root css-xxxxxxxx\\">リリースするイメージ URI と日時をセットします。</h6><h6 class=\\"MuiTypography-root MuiTypography-subtitle1 MuiTypography-root css-xxxxxxxx\\">よろしいですか？</h6><div class=\\"MuiBox-root css-xxxxxxxx\\"></div><div class=\\"MuiStack-root css-xxxxxxxx\\"><button type=\\"button\\" class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" title=\\"はい\\" tabindex=\\"0\\">はい<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button><button type=\\"button\\" class=\\"MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorInherit MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" title=\\"いいえ\\" tabindex=\\"0\\">いいえ<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></div>"`;

exports[`<Confirm /> > はい・過去リリースあり 1`] = `"<h6 class=\\"MuiTypography-root MuiTypography-subtitle1 MuiTypography-root css-xxxxxxxx\\">リリースするイメージ URI と日時をセットします。</h6><h6 class=\\"MuiTypography-root MuiTypography-subtitle1 MuiTypography-root css-xxxxxxxx\\">よろしいですか？</h6><div class=\\"MuiBox-root css-xxxxxxxx\\"></div><div class=\\"MuiStack-root css-xxxxxxxx\\"><button type=\\"button\\" class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" title=\\"はい\\" tabindex=\\"0\\">はい<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button><button type=\\"button\\" class=\\"MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorInherit MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" title=\\"いいえ\\" tabindex=\\"0\\">いいえ<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></div>"`;

exports[`<Confirm /> > はい・過去リリースなし 1`] = `"<h6 class=\\"MuiTypography-root MuiTypography-subtitle1 MuiTypography-root css-xxxxxxxx\\">リリースするイメージ URI と日時をセットします。</h6><h6 class=\\"MuiTypography-root MuiTypography-subtitle1 MuiTypography-root css-xxxxxxxx\\">よろしいですか？</h6><div class=\\"MuiBox-root css-xxxxxxxx\\"></div><div class=\\"MuiStack-root css-xxxxxxxx\\"><button type=\\"button\\" class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" title=\\"はい\\" tabindex=\\"0\\">はい<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button><button type=\\"button\\" class=\\"MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorInherit MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" title=\\"いいえ\\" tabindex=\\"0\\">いいえ<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></div>"`;
